spec/allocator_spec.rb

```diff
@@ -53,27 +53,173 @@
       expect { new_allocator(1).allocate([rate, 1 - rate]) }.not_to raise_error
     end
 
+    specify "#allocate raise ArgumentError when invalid strategy is provided" do
+      expect { new_allocator(0.03).allocate([0.5, 0.5], :bad_strategy_name) }.to raise_error(ArgumentError, "Invalid strategy. Valid options: :roundrobin, :roundrobin_reverse, :nearest")
+    end
+
+    specify "#allocate raises an error when splits exceed 100%" do
+      expect { new_allocator(0.03).allocate([0.5, 0.6]) }.to raise_error(ArgumentError, "splits add to more than 100%")
+    end
+
+    specify "#allocate scales up allocations less than 100%, preserving the relative magnitude of each chunk" do
+      # Allocations sum to 0.3
+      # This is analogous to new_allocator(12).allocate([1/3, 2/3])
+      moneys = new_allocator(12).allocate([0.1, 0.2])
```

Comment:
Nit: shouldn't this be `monies`?

Reply:
🤷🏼 

https://www.perplexity.ai/search/Is-the-correct-xEGpmiD7SqWDCO086AOu5g#0

> ## Pluralization of "Money"
> 
> The correct plural form of the word "money" is **"monies"**.
> 
> Here are the key points:
> 
> ### "Monies" vs. "Moneys"
> 
> - **"Monies"** is the standard and most common plural form of "money". This is the preferred plural in most contexts.
> - **"Moneys"** is also an accepted plural form, but it is less common and generally considered more formal or technical.
> 
> ### Usage Guidelines
> 
> - Use "monies" when referring to sums or amounts of money, especially in a general or collective sense.
>   - Example: "The organization has received significant monies in donations this year."
> - Use "moneys" when referring to specific currencies or monetary systems.
>   - Example: "The different moneys used in the European Union include the euro, the British pound, and the Swiss franc."
> 
> So in summary, "monies" is the recommended and more widely used plural form of "money" in most everyday contexts. "Moneys" is a valid alternative, but it is less common and tends to be used in more formal or specialized situations.

Reply:
thank you for the feedback, I will update to `monies`

Reply:
fixed